parser/prism/Translator.cc

```diff
@@ -1160,19 +1160,31 @@ unique_ptr<parser::Node> Translator::translate(pm_node_t *node, bool preserveCon
             ENFORCE(arg != nullptr);
             ENFORCE(arg->hasDesugaredExpr());
 
+            auto valueNode = definedNode->value;
+
             // Desugar to `::Magic.defined_instance_var(ivar)` or `::Magic.defined_class_var(cvar)`
```

Comment:
This comment applies to the two `if`s together. Can we split it?

That also gives us more room to explain that this produces a symbol:

```suggestion
            // Desugar `defined?(@ivar)` to `::Magic.defined_instance_var(:@ivar)`
```